sdk/test/UnitTests/JsonMapperTest.cs

```diff
@@ -0,0 +1,275 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="JsonMapperTest.cs" company="Amazon.com">
+//      Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+//      Licensed under the Apache License, Version 2.0 (the "License").
+//      You may not use this file except in compliance with the License.
+//      A copy of the License is located at
+//
+//      http://aws.amazon.com/apache2.0
+//
+//      or in the "license" file accompanying this file. This file is distributed
+//      on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+//      express or implied. See the License for the specific language governing
+//      permissions and limitations under the License.
+// </copyright>
+//-----------------------------------------------------------------------------
+
+using Amazon.XRay.Recorder.Core.Sampling.Model;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ThirdParty.LitJson;
+
+namespace Amazon.XRay.Recorder.UnitTests
+{
+    [TestClass]
+    public class JsonMapperTest
+    {
+        [TestMethod]
+        public void TestMarshallSamplingStatisticsDocuments()
+        {
+            var samplingStatisticsModel = new SamplingStatisticsModel();
+            var samplingStatisticsDocumentModel = new SamplingStatisticsDocumentModel
+            {
+                ClientID = "07492221d7fd13a86e750de2",
+                RuleName = "Test",
+                RequestCount = 108,
+                SampledCount = 4,
+                BorrowCount = 6,
+                Timestamp = 1604297926.362
+            };
+
+            samplingStatisticsModel.SamplingStatisticsDocuments.Add(samplingStatisticsDocumentModel);
+
+            var expected = "{\"SamplingStatisticsDocuments\":[{\"ClientID\":\"07492221d7fd13a86e750de2\",\"RuleName\":\"Test\",\"RequestCount\":108,\"SampledCount\":4,\"BorrowCount\":6,\"Timestamp\":1604297926.362}]}";
+            var actual = JsonMapper.ToJson(samplingStatisticsModel);
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        [TestMethod]
+        public void TestMarshallSamplingStatisticsDocumentsWithEmptyValues()
+        {
+            var samplingStatisticsModel = new SamplingStatisticsModel();
+            var samplingStatisticsDocumentModel = new SamplingStatisticsDocumentModel();
+
+            samplingStatisticsModel.SamplingStatisticsDocuments.Add(samplingStatisticsDocumentModel);
+
+            var expected = "{\"SamplingStatisticsDocuments\":[{\"ClientID\":null,\"RuleName\":null,\"RequestCount\":null,\"SampledCount\":null,\"BorrowCount\":null,\"Timestamp\":null}]}";
```

Comment:
What's the response from the `GetSamplingTargets` API if the sampling statistics document has null/empty required fields? Do we need to handle any exceptions in that case?

Reply:
No, they didn't do sanity check for the fields. All of the sampling statistics document [fields](https://github.com/aws/aws-sdk-net/blob/7c88eeceb52869892adb1fe50c3f06c8f1a6b126/sdk/src/Services/XRay/Generated/Model/SamplingStatisticsDocument.cs#L38-L43) can be nullable and passing a sampling statistics document object with empty values to `GetSamplingTargets` API will not throw exception.